dfusion_rust_core/src/models/account_state.rs

```diff
@@ -14,48 +15,85 @@ use super::util::*;
 pub struct AccountState {
     pub state_hash: H256,
     pub state_index: U256,
-    balances: Vec<u128>,
+    balances: HashMap<U256, HashMap<u8, u128>>, // UserId => (TokenId => balance)
```

Comment:
This looks like a complex structure: Map of a Map
Is that better than modeling this with traits that model the `AccountDetails` and allows you to get the balance for each token. It would be this way just a `HashMap<U256, AccountDetails>`

Reply:
Nice suggestion. Would be a bit of effort to expose this extra interface around it, but I agree that it'd be much cleaner.